src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java

```diff
@@ -24,4 +33,14 @@ public UserRetrievalController(UserService userService) {
     public GetAllResponse getAllUser() {
         return userService.getAllUsers();
     }
+
+    @GetMapping("/user/{userName}")
+    public ResponseEntity retrieveUser(@PathVariable("userName") @Valid String userName) {
+        User user = userService.getUserByName(userName);
+        if (user != null) {
+            return new ResponseEntity(UserResponse.builder().success(true).id(user.getId()).userName(user.getUserName()).firstName(user.getFirstName()).lastName(user.getLastName()).build(), HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>(CommonResponse.builder().success(false).errorMessage(MessagesUtil.getMessage(MessagesKeyEnum.ENTITY_USER_NOT_EXISTS.getKey())).build(), HttpStatus.NOT_FOUND);
```

Comment:
In GlobalExceptionHandler class, please make a seperate " @ExceptionHandler(NotFound.class) and make it return 404 Not found HttpStatus Code

Reply:
Removed that else part and handling exception by spring boot